Integrate `docs/bin/*.py` files into the nox config

gotmax23 opened the pull request:
This PR adds a clone-core nox session and applies the Python linters to the files in `docs/bin`.

gotmax23 committed:
add clone-core session to the noxfile

This adds a clone-core session to the noxfile that can be run with `nox
-e clone-core`.
The session is added to the default sessions in `nox.options.sessions`
so executing `nox` automatically clones core and runs all of the lint
sessions.

gotmax23 committed:
run linters over docs/bin/*.py

I ran the formatters and then fixed some line length and typing errors
manually.

oraNod reviewed (approved):
Looks perfect! Thanks @gotmax23

gotmax23 committed:
include docs/bin/*.py in noxfile linting

This enforce the Python linters in the noxfile on the Python files in
docs/bin.

gotmax23 commented:
I force pushed to split up the commits.

oraNod commented:
@nitzmahone and @mattclay Hey! Could one of you please take a look as this modifies `docs/bin`? It'd be good to confirm the changes with someone from the core team. Thanks!

oraNod commented:
doh. I may have sent review requests a little prematurely given this is in draft state.

gotmax23 commented:
> doh. I may have sent review requests a little prematurely given this is in draft state.

Nah, it should be ready for review :).

mattclay commented:
I'm not sure there's anything for me to review. The majority (all?) of the changes in `docs/bin` appear to be code formatting.

gotmax23 commented:
> I'm not sure there's anything for me to review. The majority (all?) of the changes in `docs/bin` appear to be code formatting.

The changes to those files are mostly reformatting and adding type annotations for mypy. I also removed python2-isms (`from __future__` and metaclass boilerplate; dealing with text and bytes) from find-plugins-refs.py.

gotmax23 commented:
I force pushed to fix a typo in a commit message. Other than that, I think this is good to merge.

gotmax23 commented:
Manual stable-2.15 backport: https://github.com/ansible/ansible-documentation/pull/562

oraNod commented:
> I'm not sure there's anything for me to review. The majority (all?) of the changes in `docs/bin` appear to be code formatting.

@mattclay I wanted to make sure either yourself or @nitzmahone was aware of the changes as it touched code you've added. Just hoping to avoid future surprises really. Thanks for taking a look.

oraNod commented:
Thanks for adding this @gotmax23
